pyproject.toml

```diff
@@ -32,6 +32,7 @@ boto3 = "*"
 minio = "^7.2.7"
 gevent = "^24.2.1"
 pyarrow = "16.1.0" # transitive dependency, pinned here to avoid conflicts
+flake8 = "*"
```

Comment:
I don't think we need to introduce `flake8` here?

Reply:
Yehh.., but we need it to run unit tests :(

Reply:
Ah I see, we should add it to `[tool.poetry.group.test.dependencies]` then.

Reply:
ahh good point! thanks!

Reply:
and done!